src/resource/tiles/dto/update-tile.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { IsNumber, IsObject, IsOptional, IsPositive, IsString, IsNotEmpty } from "class-validator";
+export class UpdateTileDto {
+  @IsOptional()
+  @IsNotEmpty()
+  @IsString()
+  readonly name: string;
+
+  @IsOptional()
+  @IsNotEmpty()
+  @IsObject()
+  readonly colors: object;
+
+  @IsOptional()
+  @IsNotEmpty()
+  @IsNumber()
+  @IsPositive()
+  readonly length: number;
```

Comment:
could you double check those four? not quite sure but I feel they might have some overlapping checking which means we might not need them all

Reply:
Are those properties of a tile all optional when inputing them into db?  And I agree with Chui that the props of IsOptional and IsNotEmpty are overlapped. Remove @IsOptional might be better for validation.

Reply:
It should be working as I remember there is an example like this one. It means: it is optional if you don't provide it, it's ok. But if you want to provide it in json it should not be empty.

Reply:
Yep, IsNotEmpty() is overlapping IsNumber()